### DISCUSSION TITLE
openHAB integration

### DISCUSSION THREAD

**Discussion by flodan** [Hypfer/Valetudo#901] [Q&A / Support] [answered]
I upgraded Valetudo to the `2021.04.0` version and would like to integrate it into my openHAB setup. I followed the guide on valetudo.cloud, unfortunately step 4 fails for me. The robot itself is discovered but no channels are discovered.

I use `eclipse-mosquitto` inside a docker container as MQTT broker, openHAB runs in docker as well. Am I missing any additional configuration?

Thank you very much in advance!

**Comment by depau** [comment, marked as answer]
> I upgraded Valetudo to the `2021.04.0` version and would like to integrate it into my openHAB setup. I followed the guide on valetudo.cloud, unfortunately step 4 fails for me. The robot itself is discovered but no channels are discovered.
> 
> I use `eclipse-mosquitto` inside a docker container as MQTT broker, openHAB runs in docker as well. Am I missing any additional configuration?
> 
> Thank you very much in advance!

Sometimes openHAB caches the autodiscovery info. Make sure Valetudo has published all autodiscovery topics, also go to the valetudo homepage and refresh to make sure the status is loaded.

Then restart openHAB.

If it does not autodiscover you can try to add it manually by specifying the root topic and identifier (in topic names: `ROOT_TOPIC/IDENTIFIER`).

If you already tried to add the vacuum you can follow https://valetudo.cloud/pages/integrations/openhab-integration.html#refresh-homie-channels-after-a-valetudo-upgrade

**Reply by flodan** [reply]
Restarting openHAB worked. Thank you very much!!
